src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -358,20 +355,17 @@ public void loadCube(int cubeX, int cubeY, int cubeZ, LoadType loadType, Generat
 			// ... and generating was requested, generate it.
 			if (loadType == LoadType.LOAD_OR_GENERATE) {
 				// Have the column generate a new cube object and configure it for generation.
-				cube = column.getOrCreateCube(cubeY, true);
-				cube.setCurrentStage(this.worldServer.getGeneratorPipeline().getFirstStage());
-				cube.setTargetStage(targetStage);
+				cube = this.worldServer.getCubeGenerator().generateCube(new CubeCoords(cubeX, cubeY, cubeZ), targetStage);
```

Comment:
It may be better idea for CubeGenerator to be part if ServerCubeCache (just like it is in vanilla, see `WorldServer.createChunkProvider()`).